packages/flutter/lib/src/services/text_input.dart

```diff
@@ -413,6 +417,48 @@ class TextInputConfiguration {
   /// Defaults to true.
   final bool autocorrect;
 
+  /// {@template flutter.services.textInput.enableSmartDashes}
+  /// Whether to allow the platform to automatically format dashes.
+  ///
+  /// This flag only affects iOS. It sets UITextSmartDashesType in the engine.
+  /// When true, it passes UITextSmartDashesTypeYes, and when false, it passes
+  /// UITextSmartDashesTypeNo.
+  ///
+  /// As an example of what this does, two consecutive hyphen characters will be
+  /// automatically replaced with one en dash, and three consecutive hyphens
+  /// will become one em dash.
+  ///
+  /// Defaults to true, unless obscureText is true, when it defaults to false.
+  /// This is to avoid the problem where password fields receive autoformatted
+  /// characters.
+  ///
+  /// See also:
+  ///  * [enableSmartQuotes]
```

Comment:
you need a blank line after `See also:`, see our style guide. Last sentence of <https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#leave-breadcrumbs-in-the-comments> (this matters because the rendering is broken otherwise).